objc2/src/foundation/mutable_set.rs

```diff
@@ -0,0 +1,291 @@
+use alloc::vec::Vec;
+use core::fmt;
+use core::marker::PhantomData;
+
+use super::set::with_objects;
+use super::{NSCopying, NSFastEnumeration, NSFastEnumerator, NSMutableCopying, NSObject, NSSet};
+use crate::rc::{DefaultId, Id, Owned, Ownership, Shared, SliceId};
+use crate::{ClassType, Message, __inner_extern_class, extern_methods, msg_send, msg_send_id};
+
+__inner_extern_class!(
+    /// A growable unordered collection of unique objects.
+    ///
+    /// See the documentation for [`NSSet`] and/or [Apple's
+    /// documentation][apple-doc] for more information.
+    ///
+    /// [apple-doc]: https://developer.apple.com/documentation/foundation/nsmutableset?language=objc
+    #[derive(PartialEq, Eq, Hash)]
+    pub struct NSMutableSet<T: Message> {
+        item: PhantomData<Id<T, Shared>>,
+    }
+
+    unsafe impl<T: Message> ClassType for NSMutableSet<T> {
+        #[inherits(NSObject)]
+        type Super = NSSet<T>;
+    }
+);
+
+// SAFETY: Same as NSSet<T>
+unsafe impl<T: Message + Sync + Send> Sync for NSMutableSet<T> {}
+unsafe impl<T: Message + Sync + Send> Send for NSMutableSet<T> {}
+
+extern_methods!(
+    unsafe impl<T: Message> NSMutableSet<T> {
+        /// Creates an empty `NSMutableSet`.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// use objc2::foundation::{NSMutableSet, NSString};
+        ///
+        /// let set = NSMutableSet::<NSString>::new();
```

Comment:
Tip: To get doctests to work on GNUStep:
```suggestion
        /// use objc2::foundation::{NSMutableSet, NSString};
        /// # #[cfg(feature = "gnustep-1-7")]
        /// # unsafe { objc2::__gnustep_hack::get_class_to_force_linkage() };
        ///
        /// let set = NSMutableSet::<NSString>::new();
```

Reply:
Thanks! I had no idea why doctests were failing.